plugin.video.invidious/resources/lib/invidious_api.py

```diff
@@ -0,0 +1,179 @@
+import time
+from collections import namedtuple
+
+import requests
+import xbmc
+import xbmcaddon
+import xbmcgui
+
+VideoListItem = namedtuple("VideoSearchResult",
+    [
+        "type",
+        "id",
+        "thumbnail_url",
+        "heading",
+        "author",
+        "description",
+        "view_count",
+        "published",
+        "duration",
+    ]
+)
+
+ChannelListItem = namedtuple("ChannelSearchResult",
+    [
+        "type",
+        "id",
+        "thumbnail_url",
+        "heading",
+        "description",
+        "verified",
+        "sub_count",
+     ]
+)
+
+PlaylistListItem = namedtuple("PlaylistSearchResult",
+    [
+        "type",
+        "id",
+        "thumbnail_url",
+        "heading",
+        "channel",
+        "channel_id",
+        "verified",
+        "video_count",
+    ]
+)
+
+
+class InvidiousAPIClient:
+    def __init__(self, instance_url):
+        self.instance_url = instance_url.rstrip("/")
+        self.addon = xbmcaddon.Addon()
+
+    def make_get_request(self, *path, **params):
+        base_url = self.instance_url + "/api/v1/"
+
+        url_path = "/".join(path)
+
+        while "//" in url_path:
+            url_path = url_path.replace("//", "/")
+
+        assembled_url = base_url + url_path
+
+        xbmc.log(f"invidious ========== request {assembled_url} with {params} started ==========", xbmc.LOGDEBUG)
+        start = time.time()
+        response = requests.get(assembled_url, params=params, timeout=5)
+        end = time.time()
+        xbmc.log(f"invidious ========== request finished in {end - start}s ==========", xbmc.LOGDEBUG)
+
+        if response.status_code > 300:
+            xbmc.log(f'invidious API request {assembled_url} with {params} failed with HTTP status {response.status_code}: {response.reason}.', xbmc.LOGWARNING)
+            dialog = xbmcgui.Dialog()
+            dialog.notification(
```

Comment:
In the future please move all "Notifications" and "Dialog" messages to localized language. THX

Reply:
I will try. I know how to do it with static strings.  Is there a recipe how to do this with format strings?

Reply:
#SUDO CODE#
LANG = xbmcaddon.Addon().getLocalizedString
40000 = 'foo{something}'
LANG(40000).format(something='bar')

Reply:
Fixed in the upstream code.